### PR TITLE
Command palette : Search and Recent searches plugin hook

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Related Pull Requests

- Has server changes (please link here)
- Has mobile changes (please link here)

#### Release Note

```release-note

```

### CODE REVIEW

File: actions/hooks.js [hunk 1]
```diff
@@ -89,3 +89,61 @@ export function runMessageWillBeUpdatedHooks(newPost, oldPost) {
...
+
+        const hooks = state.plugins.components.SearchInProduct;
+        if (!hooks || hooks.length === 0) {
+            return {data: []};
+        }
+
+        let searchResults = [];
+
+        for (const hook of hooks) {
+            const result = await hook.hook(searchTerm); // eslint-disable-line no-await-in-loop
```
Comment (harshilsharma63) [inline actions/hooks.js:105]: We should call this hook in parallel for each plugin.

File: actions/hooks.js [hunk 2]
```diff
@@ -89,3 +89,61 @@ export function runMessageWillBeUpdatedHooks(newPost, oldPost) {
...
+
+        const hooks = state.plugins.components.RecentInProductSearches;
+        if (!hooks || hooks.length === 0) {
+            return {data: []};
+        }
+
+        let recentSearches = [];
+
+        for (const hook of hooks) {
+            const result = await hook.hook(); // eslint-disable-line no-await-in-loop
```
Comment (harshilsharma63) [inline actions/hooks.js:134]: We should call this in parallel for each plugin.

**Review by harshilsharma63 — APPROVED** [review]
Otherwise looks good.

File: plugins/registry.js [hunk 3]
```diff
@@ -888,4 +888,38 @@ export default class PluginRegistry {
             data,
         });
     }
+
+    // Register a hook to perform search in mattermost plugins/products.
+    registerSearchInProductHook(hook) {
+        const id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
```
Comment (M-ZubairAhmed) [inline plugins/registry.js:897]: create a separate reducer for this

**Pull request merged**